early-return/starter/main.go

```diff
@@ -20,38 +21,55 @@ func main() {
 	ctxWithTimeout, cancel := context.WithTimeout(context.Background(), 5*time.Second)
 	defer cancel()
 
-	updateOperation := client.NewUpdateWithStartWorkflowOperation(
-		client.UpdateWorkflowOptions{
-			UpdateName:   earlyreturn.UpdateName,
-			WaitForStage: client.WorkflowUpdateStageCompleted,
-		})
-
-	tx := earlyreturn.Transaction{ID: uuid.New(), SourceAccount: "Bob", TargetAccount: "Alice", Amount: 100}
-	workflowOptions := client.StartWorkflowOptions{
-		ID:                 "early-return-workflow-ID-" + tx.ID,
-		TaskQueue:          earlyreturn.TaskQueueName,
-		WithStartOperation: updateOperation,
-	}
-	we, err := c.ExecuteWorkflow(ctxWithTimeout, workflowOptions, earlyreturn.Workflow, tx)
-	if err != nil {
-		log.Fatalln("Error executing workflow:", err)
-	}
+	txRequest := earlyreturn.TransactionRequest{SourceAccount: "Bob", TargetAccount: "Alice", Amount: 100}
 
-	log.Println("Started workflow", "WorkflowID", we.GetID(), "RunID", we.GetRunID())
+	startWorkflowOp := c.NewWithStartWorkflowOperation(client.StartWorkflowOptions{
+		ID:        "early-return-workflow-ID-" + uuid.New(),
+		TaskQueue: earlyreturn.TaskQueueName,
+		// WorkflowIDConflictPolicy is required when using UpdateWithStartWorkflow.
+		// Here we use FAIL, because we do not expect this workflow ID to exist already,
+		// and so we want an error if itdoes.
+		WorkflowIDConflictPolicy: enumspb.WORKFLOW_ID_CONFLICT_POLICY_FAIL,
+	}, earlyreturn.Workflow, txRequest)
 
-	updateHandle, err := updateOperation.Get(ctxWithTimeout)
-	if err != nil {
-		log.Fatalln("Error obtaining update handle:", err)
+	updateOptions := client.UpdateWorkflowOptions{
+		UpdateName:   earlyreturn.UpdateName,
+		WaitForStage: client.WorkflowUpdateStageCompleted,
 	}
 
-	err = updateHandle.Get(ctxWithTimeout, nil)
+	updateHandle, err := c.UpdateWithStartWorkflow(ctxWithTimeout, client.UpdateWithStartWorkflowOptions{
+		StartWorkflowOperation: startWorkflowOp,
+		UpdateOptions:          updateOptions,
+	})
 	if err != nil {
+		// For example, a client-side validation error (e.g. missing conflict
+		// policy or invalid workflow argument types in the start operation), or
+		// a server-side failure (e.g. failed to start workflow, or exceeded
+		// limit on concurrent update per workflow execution).
+		log.Fatalln("Error issuing update-with-start:", err)
+	}
+
+	log.Println("Started workflow",
+		"WorkflowID:", updateHandle.WorkflowID(),
+		"RunID:", updateHandle.RunID())
+
+	var txResult earlyreturn.Transaction
```

Comment:
I think that we should name this something like `earlyreturnResult`, or `transactionConfirmation`, or something else along those lines. The key point we want to get across is that the transaction hasn't completed yet, but `txResult` suggests otherwise.

Reply:
I see what you mean. I can rename it 👍 The [java-sample has the same issue](https://github.com/temporalio/samples-java/blob/main/core/src/main/java/io/temporal/samples/earlyreturn/EarlyReturnClient.java#L53C5-L53C13).